File: Sheet/Step 1/Step 1.2/python/pro1-2-17.py
# def star(n):
#     l=(2*n-1)//2
#     r=(2*n-1)//2

#     for row in range(n):
#         ch=65
        
#         for col in range(2*n-1):
            
#             if(col>=l and col<=(l+r)//2):
#                 print(chr(ch),end=' ')
#                 ch+=1

#             if(col<l or col>r):
#                 print(' ',end=' ')
            
#             if(col==(l+r)//2):
#                 ch-=1
            
#             if(col>(l+r)//2) and col<=r:
#                 ch-=1
#                 print(chr(ch),end=' ')

#         l-=1
#         r+=1
#         print()
            

def star(n):

    l=(2*n-1)//2
    r=(2*n-1)//2

    for i in range(n):
        ch=65

        for j in range(2*n-1):

            if(j>=l and j<=(l+r)//2):
                print(chr(ch),end= ' ')
                ch+=1

            if(j<l or j>r):
                print(' ', end=' ')

            if(j==(l+r)//2):
                ch-=1

            if(j>(l+r)//2) and j<=r:
                ch-=1
                print(chr(ch),end=' ')

        l-=1
        r+=1
        print()


            


z = 6
star(z)

File: Sheet/Step 1/Step 1.2/C++/pro1-2-17.cpp
#include <iostream>
using namespace std;

int main()
{
    int a = 4; // Define the number of rows

    for (int row = 0; row < a; row++)
    {
        // Loop for each row from 0 to 'a' - 1
        for (int col = 0; col < a - row - 1; col++)
        {
            // Loop to print spaces before the pattern
            cout << " ";
        }

        char ch = 'A';              // Initialize character 'A'
        int bp = (2 * row + 1) / 2; // Calculate the breakpoint for changing character direction
        for (int col = 1; col <= 2 * row + 1; col++)
        {
            // Loop to print the pattern characters
            cout << ch; // Print the character

            if (col <= bp)
            {
                // If the current column is before the breakpoint, increment the character
                ch++;
            }
            else
            {
                // Otherwise, decrement the character
                ch--;
            }
        }

        for (int col = 0; col < a - row - 1; col++)
        {
            // Loop to print spaces after the pattern
            cout << " ";
        }
        cout << endl; // Move to the next line after printing the pattern for the current row
    }
    return 0; // Indicate successful execution of the program
}
